xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-webjar/src/main/vue/displayers/BaseDisplayer.vue

```diff
@@ -28,61 +28,67 @@
 -->
 <template>
   <div
-    :class="isView ? 'view' : 'edit'"
-    @dblclick="edit"
-    @keypress.self.enter="edit"
-    :tabindex="isEditable ? 0 : ''"
+      :class="isView ? 'view' : 'edit'"
+      @dblclick="toggleIsView"
+      @keypress.self.enter="toggleIsView"
+      :tabindex="isEditable ? 0 : ''"
   >
     <!--
-      The base displayer contains two slots: `viewer` and `editor`.
-      It displays the one according to its current state: `this.isView`.
+      The base displayer contains three slots: `viewer`, `editor`, and `loading`.
+      It displays `viewer` or `loading` according to its current state: `this.isView` when `this.isLoading` is false,
+      and `loading` otherwise.
     -->
 
     <!-- The slot containing the displayer Viewer widget -->
     <slot
-      name="viewer"
-      v-if="isView"
+        name="viewer"
+        v-if="isView && !isLoading"
     >
       <!--
         Default Viewer widget
         Normally this should rarely be used, as the default displayer
         should provide a default viewer if no displayer is specified
-        However, this is usefull if a custom displayer only implement
+        However, this is useful if a custom displayer only implement
         its Editor widget, as a default Viewer widget would still be provided
       -->
       <div>{{ value }}</div>
     </slot>
 
     <!-- The slot containing the displayer Editor widget -->
     <slot
-      name="editor"
-      v-if="!isView"
+        name="editor"
+        v-if="!isView && !isLoading"
     >
       <!--
         Default Editor widget
         Normally this should rarely be used, as the default displayer
         should provide a default editor if no displayer is specified
-        However, this is usefull if a custom displayer only implement
+        However, this is useful if a custom displayer only implement
         its Viewer widget, as a default Editor widget would still be provided
       -->
       <input
-        class="default-input"
-        type="text"
-        size="1"
-        v-autofocus
-        :value="value"
-        @focusout="applyEdit($event.target.value)"
-        @keypress.enter="applyEdit($event.target.value)"
-        @keydown.esc="cancelEdit"
+          class="default-input"
+          type="text"
+          size="1"
+          v-autofocus
+          :value="value"
+          @focusout="applyEdit($event.target.value)"
+          @keypress.enter="applyEdit($event.target.value)"
+          @keydown.esc="cancelEdit"
       />
     </slot>
 
+    <slot name="loading" v-if="isLoading">
+      <XWikiLoader></XWikiLoader>
+    </slot>
```

Comment:
currently unused, but should be used to show a loading content while the form is saved or loaded